### DISCUSSION TITLE
about names of generated sources

### DISCUSSION THREAD

**Discussion by jwuttke** [RosettaCommons/binder#187] [General]
Running `binder --root-module PRJ --prefix auto [...] all_includes.hpp` where `all_includes.hpp` is
```
#include <A.hpp>
#include <B.hpp>
#include <C.hpp>
```
directory `auto` will contain
```
A.cpp
PRJ.cpp
```
Why this arbitrary choice of file name `A.cpp`, taken from the first include. Why not rather `all_includes.cpp`?

**Comment by lyskov** [comment]
When possible Binder is try to place generated code into file named after include file declaration comes from. In principle this allows generated sources to mimic dir structure of original project. Main reason for this is to simplify inspection of generated code or large projects by human when desired. For example if we know that class `AAA` was declared in `A.hpp` then most likely generated source code will be located in file `A.cpp`. This will be particular useful when original project is large and have complex namespace layout with multiple files/functions with identical names in different classes/namespaces.

Also, if primary use case is to generate bindings for tiny projects (say <10K of LOC) then using Binder `--single-file` option might be best option. When this option is specified all generated code will be placed into single output file named after root-module.
